### PR TITLE
Feat: 2126 Upload verification files

### PR DESCRIPTION
**Pull request by BCerki**
Card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-ciip-portal/2126

### CODE REVIEW

File: app/server/postgraphile/uploadFieldPlugin.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/* eslint-disable */
```
Comment (BCerki) [inline app/server/postgraphile/uploadFieldPlugin.js:1]: This is copy-pasted from CIF with no linting errors, so I'm assuming it's fine to disable here

File: helm/cas-ciip-portal/templates/bucket-provisioning-job.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.db.restore.enabled }}
```
Comment (BCerki) [inline helm/cas-ciip-portal/templates/bucket-provisioning-job.yaml:1]: Note to self: delete this whole file, already handled in another PR

File: app/containers/Forms/VerificationStatement.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            return (
+              <>
+                <a href="#" className="attachment-link" key={node.id}>
+                  {node.fileName}
+                </a>{" "}
+                <FontAwesomeIcon
+                  icon={faTrash}
+                  onClick={() => {
+                    const { environment } = relay;
+                    return deleteAttachmentMutation(environment, {
```
Comment (dleard) [inline app/containers/Forms/VerificationStatement.tsx:102]: **To be done in a separate PR:** 
This deletes the attachment from the database, but not the GCS bucket. We will need to use https://www.npmjs.com/package/@google-cloud/storage to delete the file from GCS & then remove it from the database if the GCS delete was successful.

File: app/containers/Forms/VerificationStatement.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        },
+      },
+    };
+
+    await createAttachmentMutation(environment, variables)
+      .catch(() => {
+        setIsUploading(false);
+      })
+      .then(() => setIsUploading(false));
+  };
```
Comment (dleard) [inline app/containers/Forms/VerificationStatement.tsx:59]: **To be done in a separate PR:**
We should put in some protections on file size / type. Let's reach out to CAS & determine if they know what file types they expect these verification statements to be formatted as.

File: app/mutations/application/deleteAttachment.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { graphql } from "react-relay";
+import RelayModernEnvironment from "relay-runtime/lib/store/RelayModernEnvironment";
+import {
+  deleteAttachmentMutationVariables,
+  deleteAttachmentMutation as deleteAttachmentMutationType,
+} from "deleteAttachmentMutation.graphql";
+import BaseMutation from "mutations/BaseMutation";
+
+const mutation = graphql`
+  mutation deleteAttachmentMutation(
```
Comment (dleard) [inline app/mutations/application/deleteAttachment.ts:10]: **To be done in a separate PR:**
We will want to handle delete differently for previous versions (hard delete if attachment is from current version, soft delete it attachment was added in a previous version).
We can either define two mutations here and conditionally pass them to the performMutation() call below, or we can override the behaviour with a custom SQL delete mutation.

File: schema/deploy/tables/attachment.sql [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+create index attachment_file on ggircs_portal.attachment(file);
+create index application_id on ggircs_portal.attachment(application_id);
+create index ggircs_portal_attachment_foreign_key on ggircs_portal.attachment(application_id, version_number);
+
+select ggircs_portal_private.upsert_timestamp_columns(
+  table_schema_name := 'ggircs_portal',
+  table_name := 'attachment',
+  add_create := true,
+  add_update := true,
+  add_delete := false);
```
Comment (dleard) [inline schema/deploy/tables/attachment.sql:28]: **To be done in a separate PR:**
We will need to make `add_delete` true here to enable soft delete

Comment (dleard) [reply schema/deploy/tables/attachment.sql:28]: (And add comments on the new `delete` columns)

**Review by dleard — APPROVED** [review]
Great work on this! 
There are a few things that we need to address around upload/delete but I am approving this PR as-is since it unblocks the download card #2181 & allows us to do more work in parallel. 
I'll write up a new ticket in zenhub to capture the changes I mentioned in the comments on this PR.
